docs/set-up.asciidoc

```diff
@@ -39,32 +40,278 @@ include::./custom-stack.asciidoc[]
+
+[source,js]
+----
+import 'elastic-apm-node/start';
+
+// Application main code goes here.
+----
+
+A limitation of this approach is that you cannot configure the agent with an options object, but instead have to rely on <<configuring-the-agent,one of the other methods of configuration>>, such as setting `ELASTIC_APM_...` environment variables.
+
+
+[[start-option-node-require-opt]]
+===== `node -r elastic-apm-node/start ...`
+
+Another way to start the agent is with the `-r elastic-apm-node/start` https://nodejs.org/api/cli.html#-r---require-module[command line option to node]. This will import and start the APM agent before your application code starts. This method allows you to enable the agent _without touching any code_. This is the recommended start method for <<lambda,monitoring AWS Lambda functions>>.
+
+[source,bash]
+----
+node -r elastic-apm-node/start app.js
+----
+
+The `-r, --require` option can also be specified via the https://nodejs.org/api/cli.html#node_optionsoptions[`NODE_OPTIONS` environment variable]:
+
+[source,bash]
+----
+# export ELASTIC_APM_...  # Configure the agent with envvars.
+export NODE_OPTIONS='-r elastic-apm-node/start'
+node app.js
+----
+
+
+[[start-option-separate-init-module]]
+===== separate APM init module
+
+If you want to avoid <<start-esm-imports,the gotcha with hoisted ES modules>> but still want the flexibility of passing a config object to the <<apm-start,agent start method>>, then a good option is to write a separate JavaScript or TypeScript module that starts the agent, and import *that* init module at the top of your main file. For example:
+
+[source,ts]
+----
+// initapm.ts
+import apm from 'elastic-apm-node';
+apm.start({
+  serverUrl: 'https://...',
+  secretToken: '...',
+  // ...
 })
 ----
-* Require the `elastic-apm-node/start` script:
-+
+
+[source,ts]
+----
+// main.ts
+import 'initapm'
+
+// Application code starts here.
+----
+
+
+[[start-gotchas]]
+==== Start Gotchas
+
+This section shows some sometimes subtle surprises starting the APM agent with some technologies. A general troubleshooting tip for using the Elastic Node.js APM agent with any build tool/system that produces compiled JavaScript is to **look at the compiled JavaScript** to see what is actually being executed by `node`.
+
+[[start-esm-imports]]
+===== Hoisted ES module imports
+
+When using a compiler/transpiler (such as https://babeljs.io/docs/en/[Babel] or https://esbuild.github.io[esbuild]) to translate from code using ES modules (`import ...` statements) to CommonJS modules (`require(...)`) there is a common surprise for starting the Elastic Node.js APM agent. Running Babel on the following produces code that does *not* work as expected:
```

Comment:
So the "gotcha" here is that imports are hoisted to the top of a module. But in the intro, we only call it a "common surprise". I wonder if the hoisted explanation should be moved to the top so that we immediately explain the meat of the issue. For example:


> When using a tool like Babel or esbuild to translate/transpile from code using ES modules (i.e. `import ...` statements) to code using CommonJS (i.e. `require(...)`), all imports are "hoisted" to the top of a module, properly following ECMAScript module (ESM) semantics. This means the `apm.start()` method is called too late—*after* the `http` module has been imported.

> For example, running Babel on the following code does not initiate APM early enough:

> **example and equivalent example**

> The <<start-option-require-start-module,the `elastic-apm-node/start` module>> fixes this problem. The following will work:

> **example**

> A more complete example...

Reply:
Done in commit c9fb5c34